sql/core/src/main/java/org/apache/spark/sql/execution/vectorized/ColumnVector.java

```diff
@@ -175,9 +175,7 @@ public ColumnarRow getStruct(int rowId, int size) {
    * Returns the array at rowid.
    */
   public final ColumnarArray getArray(int rowId) {
-    resultArray.length = getArrayLength(rowId);
-    resultArray.offset = getArrayOffset(rowId);
-    return resultArray;
+    return new ColumnarArray(arrayData(), getArrayOffset(rowId), getArrayLength(rowId));
```

Comment:
Is it better to create `ColumnarArray` for each `rowID` only once (e.g. by using caching)? I am curious whether we would see performance overhead for creating `ColumnarArray` to access elements of a multi-dimensional array (e.g. `a[1][2] + a[1][3]`).

Reply:
I don't think that is a good idea. That would require us to keep an array of `ColumnarArray` around. That might mess with both GC and escape analysis. Let's just create a benchmark and check if we do not regress.